core/src/main/scala/org/apache/spark/sql/delta/commands/OptimizeTableCommand.scala

```diff
@@ -237,7 +237,7 @@ class OptimizeExecutor(
       sparkSession.sparkContext.getLocalProperty(SPARK_JOB_GROUP_ID),
       description)
 
-    val addFiles = txn.writeFiles(repartitionDF).collect {
+    val addFiles = txn.writeFiles(repartitionDF, actionType = "Optimize").collect {
```

Comment:
I haven't tried, but wondering if there is a way we could specify the `SortOrder` on the `repartionDf` here itself, so that we can avoid passing a action type to `txn.writeFiles`?

Reply:
I think no way to do that as it's not a spark plan. Also Spark Optimizer might build some incorrect plan with a fake df, so it's better to add after applying all rules.
We may need to add the actionType for OptimizeWrite, to exclude OPTIMIZE/ZORDER/Auto compaction for OptimizeWrite feature after all.

Reply:
Maybe we can read actionType from TahoeBatchFileIndex.
Let me try it.

Reply:
I tried this approach, but it requires some code in OptimizeTableCommand.scala - extracting physcalPlan for partitionColumns/outputattributes... etc
In addition, I'd like to minimize the side effect that may incur from the fake outputOrdering while spark optimizer. Though I know the plan for compaction is not that complicated. 

How about the current PR? 
In any case I need to use TransactionalWrite.isOptimizeCommand for OptimizeWrite functionality.